geopandas/tools/sjoin.py

```diff
@@ -302,9 +308,12 @@ def _frame_join(join_df, left_df, right_df, how, lsuffix, rsuffix):
                 right_index=True,
                 suffixes=("_{}".format(lsuffix), "_{}".format(rsuffix)),
             )
-            .set_index(index_left)
             .drop(["_key_right"], axis=1)
         )
+        try:
+            joined = joined.set_index(index_left)
+        except KeyError:
+            joined = joined.set_index(index_left + "_{}".format(lsuffix))
```

Comment:
This adds some complexity to need to add this try/except here in all cases, but it should only be temporary, as I plan to simplify all of this a lot by using concat instead of merge (after the 0.10 release :))